### PR TITLE
Firecracker release v0.20.0

### LINKED ISSUE #1283: Can't start a VM in AARCH64 with vcpus number more than 16.
When start a VM in AARCH64 with more than 16 vcpus, the VM can start but only 16 vcpus can be online in VM.
For example, I started a VM with 32 vcpus.
```
[    0.022545] psci: failed to boot CPU16 (-22)
[    0.047729] CPU16: failed to boot: -22
[    0.048078] CPU16: failed in unknown state : 0x0
[    0.048723] psci: failed to boot CPU17 (-22)
[    0.049122] CPU17: failed to boot: -22
[    0.049473] CPU17: failed in unknown state : 0x0
[    0.050170] psci: failed to boot CPU18 (-22)
[    0.050564] CPU18: failed to boot: -22
[    0.050902] CPU18: failed in unknown state : 0x0
[    0.051534] psci: failed to boot CPU19 (-22)
[    0.051943] CPU19: failed to boot: -22
[    0.052297] CPU19: failed in unknown state : 0x0
[    0.052943] psci: failed to boot CPU20 (-22)
[    0.053340] CPU20: failed to boot: -22
[    0.053691] CPU20: failed in unknown state : 0x0
[    0.054364] psci: failed to boot CPU21 (-22)
[    0.054767] CPU21: failed to boot: -22
[    0.055119] CPU21: failed in unknown state : 0x0
[    0.055809] psci: failed to boot CPU22 (-22)
[    0.056203] CPU22: failed to boot: -22
[    0.056543] CPU22: failed in unknown state : 0x0
[    0.057169] psci: failed to boot CPU23 (-22)
[    0.057570] CPU23: failed to boot: -22
[    0.057936] CPU23: failed in unknown state : 0x0
[    0.058605] psci: failed to boot CPU24 (-22)
[    0.059021] CPU24: failed to boot: -22
[    0.059381] CPU24: failed in unknown state : 0x0
[    0.060052] psci: failed to boot CPU25 (-22)
[    0.060463] CPU25: failed to boot: -22
[    0.060811] CPU25: failed in unknown state : 0x0
[    0.061419] psci: failed to boot CPU26 (-22)
[    0.061819] CPU26: failed to boot: -22
[    0.062170] CPU26: failed in unknown state : 0x0
[    0.062840] psci: failed to boot CPU27 (-22)
[    0.063241] CPU27: failed to boot: -22
[    0.063586] CPU27: failed in unknown state : 0x0
[    0.064254] psci: failed to boot CPU28 (-22)
[    0.064650] CPU28: failed to boot: -22
[    0.064985] CPU28: failed in unknown state : 0x0
[    0.065605] psci: failed to boot CPU29 (-22)
[    0.066021] CPU29: failed to boot: -22
[    0.066358] CPU29: failed in unknown state : 0x0
[    0.066995] psci: failed to boot CPU30 (-22)
[    0.067396] CPU30: failed to boot: -22
[    0.067753] CPU30: failed in unknown state : 0x0
[    0.068415] psci: failed to boot CPU31 (-22)
[    0.068816] CPU31: failed to boot: -22
[    0.069169] CPU31: failed in unknown state : 0x0

```
After boot into VM,  use `lscpu`. The vcpu 0-15 is online and 16-31 is offline. BTW,  I can't make them offline.

### LINKED ISSUE #1316: Mounting block devices as virtio devices doesn't work correctly
While investigating https://github.com/firecracker-microvm/firecracker-containerd/issues/253, we have found a small gap regarding virtio block device support on Firecracker.

Currently Firecracker is using `std::fs::metadata()` to [check the size of a virtio block device](https://github.com/firecracker-microvm/firecracker/blob/4fea9ad1cc9eb15776b5880bfc67296f9bb036ca/vmm/src/lib.rs#L1910), which ultimately calls [stat() on Unix](https://github.com/rust-lang/rust/blob/2d1a551e144335e0d60a637d12f410cf65849876/src/libstd/sys/unix/fs.rs#L522).

However firecracker-containerd is using a loopback device as the backing file for a virtio block device. On a lookback device, the size from stat() is always zero.

Can Firecracker support our usecase by inspecting the size of the file more deeply? You probably can use [BLKGETSIZE64 ioctl](http://www.microhowto.info/howto/get_the_size_of_a_linux_block_special_device_in_c.html).

Thanks,

### LINKED ISSUE #1375: set logger with null options will crash firecracker
The following logger configure would crash firecracker:
```
 curl --unix-socket /tmp/firecracker.socket  -i     -X PUT "http://localh
ost/logger"     -H "accept: application/json"     -H "Content-Type: application/json"     -d "{
             \"log_fifo\": \"/logs.fifo\",
             \"metrics_fifo\": \"/metrics.fifo\", \"level\": \"Debug\", \"options\":null
    }"
```

Firecracker will crashed with the following backtrace:

2019-11-08T14:24:57.361247895 [anonymous-instance] Firecracker panicked at 'called `Option::unwrap()` on a `None` value', /rustc/50f8aadd746ebc929a752e5ffb133936ee75c52f/src/libcore/macros.rs:41:40
2019-11-08T14:24:57.510482482 [anonymous-instance] stack backtrace:
   0: backtrace::backtrace::libunwind::trace::h952205ac5c19428a (0x8a16ad)
             at /home/fupan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/libunwind.rs:88
      backtrace::backtrace::trace_unsynchronized::hdd58e47a2f7fc4c0
             at /home/fupan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/mod.rs:66
   1: backtrace::backtrace::trace::hf08ff4662e0235f8 (0x8a1665)
             at /home/fupan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/mod.rs:53
Aborted

### PR DESCRIPTION
**Pull request by serban300**
## Reason for This PR

Firecracker release v0.20.0

## Description of Changes

```
### Added

- Added support for GICv2.

### Fixed

- Fixed a logical error in bounds checking performed on vsock virtio
  descriptors (CVE-2019-18960).
- Fixed #1283 - Can't start a VM in AARCH64 with vcpus number more than 16.
- The backtrace are printed on `panic`, no longer causing a seccomp fault.
- Fixed #1375 - Change logger options type from Value to Vec<LogOption> to
  prevent potential unwrap on None panics.
- Raise interrupt for TX queue used descriptors - Github issue #1436
- Fixed a bug that causes 100% cpu load when the net device rx is throttled
  by the ratelimiter - Github issue #1439
- Invalid fields in rate limiter related API requests are now failing with
  a proper error message.
- Fixed #1316 - fix the size of a virtio device, backed by a block device.
- Log failed api requests.

### Changed

- Decreased release binary size by 10%.
```

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.
- [x] Either no new `unsafe` code has been added, or, the newly added `unsafe`
      code is unavoidable and properly documented.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+- Added support for GICv2.
 
 ### Fixed
 
 - Fixed #1283 - Can't start a VM in AARCH64 with vcpus number more than 16.
 - The backtrace are printed on `panic`, no longer causing a seccomp fault.
 - Fixed #1375 - Change logger options type from Value to Vec<LogOption> to
   prevent potential unwrap on None panics.
+- Fixed #1316 - fix the size of a virtio device, backed by a block device
+- api: log failed api requests
```
[the remark this answers is not in the document]
Comment (serban300) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
 # Changelog
 
-## [Unreleased]
+## [0.20.0]
+
+### Added
```
Comment (iulianbarbu) [inline CHANGELOG.md]: Should we also mention the UPD support from dumbo? (https://github.com/firecracker-microvm/firecracker/commit/03e61c5d7088aaa4612e6bfa984490f75acfb3a2)

Comment (serban300) [reply CHANGELOG.md]: I don't think so. Doesn't seem customer facing.

**Comment by andreeaflorescu** [comment]
Can we resume the work on this release? All relevant blockers seem to be merged now.

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +19,12 @@
   by the ratelimiter - Github issue #1439
 - Invalid fields in rate limiter related API requests are now failing with
```
Comment (dianpopa) [inline CHANGELOG.md]: nit: This is also related to a issue #1437.

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +19,12 @@
...
   a proper error message.
+- Fixed #1316 - fix the size of a virtio device, backed by a block device.
+- Log failed api requests.
```
Comment (dianpopa) [inline CHANGELOG.md]: Is this #1383?

File: CHANGELOG.md (hunk 3 above)
Comment (acatangiu) [reply CHANGELOG.md]: I agree we should add the issue numbers where applicable (like here). Also +1 on consistent style when referencing issue numbers.

Comment (serban300) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md (hunk 4 above)
Comment (serban300) [reply CHANGELOG.md]: Yes. I added a reference to it.

**Pull request merged**